src/provider/index.ts

```diff
@@ -75,12 +75,22 @@ export class Main {
                 const {notifications} = action;
                 notifications.forEach((storedNotification: StoredNotification) => {
                     const {notification, source} = storedNotification;
-                    const event: Targeted<Transport<NotificationClosedEvent>> = {
+                    if (notification.onClose) {
```

Comment:
Just noticed these checks are a bit inconsistent... The docs say that we'll treat any non-undefined value as an action. Though the pre-release change we made to the types only expects objects.

Should we standardise on `if (<action> !== undefined) {` for these checks?

Reply:
I think yes. Though ugh, means `null` will raise an action.

Reply:
Actually, we're inconsistent here. On the notification itself we have:
`onSelect?: ActionDeclaration<never, never>|null`
`onClose?: ActionDeclaration<never, never>|null;`

(With the docs saying "If omitted or `null`")

But on the button we have:

`onClick?: ActionDeclaration<never, never>;`

Which way should we go here?

Reply:
Have the hydration fill-in unspecified `on*` properties with `null`, and change these checks to `if (notification.on* !== null) {`?

Reply:
Though, should we drop the `| null` from `onSelect` and `onClose`. Can we do that?

Reply:
Discussed - will add `| null` to `onClick`, and check how buttons are hydrated

Reply:
Done, but needed use of `DeepRequired` from util-types module. Should run by you when you've got time

✅

Reply:
Actually, don't look just yet...